src/analyze_rounding.rs

```diff
@@ -94,8 +93,8 @@ fn visit(expr: &Expr, rounding_direction: bool, config: &mut config::Config) ->
             let (left_rounding, right_rounding) = match op {
                 Opcode::Add => (rounding_direction, rounding_direction),
                 Opcode::Sub => (rounding_direction, !rounding_direction),
-                Opcode::Mul(op_rounding) => handle_mul(rounding_direction, op_rounding),
-                Opcode::Div(op_rounding) => handle_div(rounding_direction, op_rounding),
+                Opcode::Mul(mut op_rounding) => handle_mul(rounding_direction, &mut op_rounding),
+                Opcode::Div(mut op_rounding) => handle_div(rounding_direction, &mut op_rounding),
```

Comment:
I don't fully understand the logic, but these lines give me pause.

With the changes, `Rounding` derives `Copy`, so the `op_rounding` on the right of the `=>` will be a copy of the one in the `Opcode`.

So then `handle_mul`/`handle_div` will modify that copy, not the one in the `Opcode`.

This seems like a functional change from the original. Or have I misunderstood?